File: frontend/src/types/resume-template.ts
export interface ResumeTemplate {
  id: string
  name: string
  yaml_content: string
  is_master: boolean
  is_starred: boolean
  created_at: string
  updated_at: string
  linked_application_count: number
}

export interface ResumeTemplateDetail extends ResumeTemplate {
  linked_applications: LinkedApplicationSummary[]
}

export interface LinkedApplicationSummary {
  id: string
  job_title: string | null
  company: string | null
  status: string
}

export interface ResumeTemplateCreate {
  name: string
  yaml_content?: string
}

export interface ResumeTemplateUpdate {
  name?: string
  yaml_content?: string
  is_starred?: boolean
}

File: backend/app/schemas/resume_template.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional


class ResumeTemplateCreate(BaseModel):
    name: str
    yaml_content: Optional[str] = None  # If None, will clone from master


class ResumeTemplateUpdate(BaseModel):
    name: Optional[str] = None
    yaml_content: Optional[str] = None
    is_starred: Optional[bool] = None


class ResumeTemplateYamlUpdate(BaseModel):
    yaml_content: str


class LinkedApplicationSummary(BaseModel):
    id: str
    job_title: str
    company: str
    status: str

    model_config = ConfigDict(from_attributes=True)


class ResumeTemplateResponse(BaseModel):
    id: str
    name: str
    yaml_content: str
    is_master: bool
    is_starred: bool
    created_at: datetime
    updated_at: datetime
    linked_application_count: int = 0

    model_config = ConfigDict(from_attributes=True)


class ResumeTemplateDetailResponse(ResumeTemplateResponse):
    linked_applications: list[LinkedApplicationSummary] = []


class ApplicationStatusUpdate(BaseModel):
    status: str


class ApplicationResumeTemplateUpdate(BaseModel):
    resume_template_id: Optional[str] = None
